zkevm-circuits/src/table.rs

```diff
@@ -444,28 +444,17 @@ pub struct RwTable {
     /// Key1 (Id)
     pub id: Column<Advice>,
     /// Key2 (Address)
-    pub address: Column<Advice>,
+    pub address: word::Word<Column<Advice>>,
```

Comment:
We shouldn't turn this into a word. As the spec says, the address column is used for multiple purposes. Let me quote them below.

https://github.com/privacy-scaling-explorations/zkevm-specs/blob/master/specs/tables.md#rw_table

>  - col. 4 *Address* is the position to **Stack**, **Memory**, or account, where the read or write takes place, depending on the cell value of the *Tag* column.
        - If the *Tag* value is "Account", the cell represents 160 bits **address**.
        - If the *Tag* value is "Stack", the cell represents 10 bits  **stackPointer**.
        - If the *Tag* value is "Memory", the cell represents 32 bits **memoryAddress**.
        - If the *Tag* value is "TxLog", then the cell represents the packing of 2 values:
            - **logID**: 32 bits, starts at 1 (corresponds to `logIndex + 1`), unique per tx/receipt.
            - **topicIndex, byteIndex**: 32 bits, indicates order in tx log topics or data.

Most of the usages can fit into a cell. The only usage that might be a word is the "Eth address". Since lots of our API returns address as a word, we might tempt to change Rwtable's address column into word columns. But considering the other use cases, changing the address to word columns adds confusion to the stackPointer and Memory address use cases.

Reply:
This is quite an open question and discussed in another pr. I also vote +1  to have just single cell in `rwtable.address` since it's used for multi-purposes. and also benefit to save one extra `hi` cell for lookup.